src/SDKs/DataFactory/Management.DataFactory/Generated/Models/AmazonS3Dataset.cs

```diff
@@ -31,7 +30,8 @@ public partial class AmazonS3Dataset : Dataset
         /// </summary>
         public AmazonS3Dataset()
         {
-          CustomInit();
+            LinkedServiceName = new LinkedServiceReference();
```

Comment:
Performing this initialization looks like a bug in the new version of AutoRest, since this is done in the base class constructor. @fearthecowboy may be interested 😄

Reply:
Hmm..

@olydis  -- perhaps this is a side effect of the work you did with polymorphic deserialization fixes?

Reply:
@fearthecowboy @bgold09 can't tie it to recent changes I'm aware of *BUT* now that you say it:
This occurrence right here isn't really a problem right? Sure, the base class ctor also initializes it, so now it happens twice.
But look down to [line 60](https://github.com/Azure/azure-sdk-for-net/pull/3754/files/2267cb316aeb088779d3dafb944a4f35909592d9#diff-c47832f618498117361e79e9e75e2820R60)! That ctor takes a `linkedServiceName` and passes it to the base class ctor (which I assume takes that value for initialization) - and then it gets re-initialized! So whatever gets passed is overwritten right after! How did this behavior not cause trouble so far?

Reply:
@bgold09 -- What version of autorest did you generate the previous version with ?  and was it the exact same swagger? I'm worried we have an unintentional change in the generator here.

Reply:
@fearthecowboy yes, this was generated with the [same version of the spec as before](https://github.com/Azure/azure-rest-api-specs/tree/current/specification/datafactory/resource-manager), using AutoRest 1.2.2. 

@olydis you're right, I don't think this ever worked and presumably no one caught it on our end! Any idea why this is being generated that way? Definitely looks like an issue related to polymorphism and potentially also due to flattening

Reply:
@fearthecowboy @bgold09 tracking down the issue, it seems that AutoRest mistakenly thinks `LinkedServiceReference` is a constant type, i.e. all of its properties are constant, so it can blindly initialize it in the constructor instead of taking it as a parameter. However, only the part that generates the code *within* the method makes that wrong decision, the part responsible for the *signature* still detects the type as non-constant. You should see the problem go away when you remove the `enum` restriction from the one property of `LinkedServiceReference`. I'll try to track down the issue and work on a fix ASAP so this can be regened.

Reply:
@olydis removing the enum from the property type of LinkedServiceReference would not be appropriate, since that makes the type get generated in such a way that it type gets de/serialized properly. Consumers of the SDK are not allowed to change that value

Reply:
@bgold09 oh sorry, I wasn't trying to suggest that you make that edit, that's just the key detail I identified as triggering the bug... as mentioned, I'm fixing it. I'll give you an additional command line switch for `autorest` that you can test in a few minutes.

Reply:
@bgold09 can you try adding `--use=https://github.com/olydis/scratch/raw/master/microsoft.azure-autorest.csharp-bgold09.tgz` to the `autorest` call? Let me know if that generates better looking code for you ;)

Reply:
I had a feeling that was what you meant, sorry about that! Let me give that a shot

Reply:
That fixed the issue, so whatever change you made worked!

Reply:
awesome, thanks for testing!

Reply:
Any thoughts on when a new csharp extension might be published with this fix? Not sure what your typical release cadence is

Reply:
hmmm, whenever we're confident enough that I didn't accidentally mess anything up. Latest: 2nd Wednesday this month

Reply:
since the extension is now published (as preview), you should be able to also go `--use=@microsoft.azure/autorest.csharp@preview` instead of referencing some voodoo URL ;)